### PR TITLE
feat(sso):add sso web and auth admin web login

### PR DESCRIPTION
**Pull request by MayueCif**

### CODE REVIEW

File: src/Web/Masa.Auth.Web.Sso/Pages/Grants/ViewModel.cs [hunk 1]
```diff
@@ -0,0 +1,22 @@
+// Copyright (c) MASA Stack All rights reserved.
+// Licensed under the Apache License. See LICENSE.txt in the project root for license information.
+
+namespace Masa.Auth.Web.Sso.Pages.Grants;
+
+public class ViewModel
```
Comment (doddgu) [inline src/Web/Masa.Auth.Web.Sso/Pages/Grants/ViewModel.cs:6]: GrantViewModel

File: src/Web/Masa.Auth.Web.Sso/Pages/Grants/ViewModel.cs [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+{
+    public IEnumerable<GrantViewModel> Grants { get; set; } = new List<GrantViewModel>();
+}
+
+public class GrantViewModel
```
Comment (doddgu) [inline src/Web/Masa.Auth.Web.Sso/Pages/Grants/ViewModel.cs:11]: GrantDetailViewModel

File: src/Web/Masa.Auth.Web.Sso/Pages/_HostAuthModel.cs [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+                    {
+                        _ssoAuthenticationStateCache.AddAuthorizationContext(returnUrl, await _identityServerInteractionService.GetAuthorizationContextAsync(returnUrl));
+                    }
+                }
+                break;
+            case "/grants":
+            case "/grants/index":
+                _ssoAuthenticationStateCache.Grants = await _identityServerInteractionService.GetAllUserGrantsAsync();
+                break;
+        }
```
Comment (doddgu) [inline src/Web/Masa.Auth.Web.Sso/Pages/_HostAuthModel.cs:43]: default?

File: src/Web/Masa.Auth.Web.Sso/Pages/_Layout.cshtml [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+  @RenderBody()
+  <script src="_framework/blazor.server.js"></script>
+  <script src="_content/BlazorComponent/js/blazor-component.js"></script>
+
+  <script>
+    window.login = (data) => {
+      const Http = new XMLHttpRequest();
+      const url = 'login';
+      Http.open("post", url, false);
+      //发送合适的请求头信息
```
Comment (doddgu) [inline src/Web/Masa.Auth.Web.Sso/Pages/_Layout.cshtml:33]: English

File: src/Web/Masa.Auth.Web.Sso/Program.cs [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+builder.Services.AddGlobalForServer();
+
+builder.Services.AddHealthChecks();
+
+builder.Services.AddMasaIdentityModel(IdentityType.MultiEnvironment);
+builder.Services.AddAuthClient(builder.Configuration.GetValue<string>("AuthClient:Url"));
+builder.Services.AddPmClient(builder.Configuration.GetValue<string>("PmClient:Url"));
+
+builder.Services.AddOidcCacheStorage(builder.Configuration.GetSection("RedisConfig").Get<RedisConfigurationOptions>())
+.AddIdentityServer(options =>
```
Comment (doddgu) [inline src/Web/Masa.Auth.Web.Sso/Program.cs:28]: Code foramt.

File: src/Web/Masa.Auth.Web.Sso/Shared/SideLayout.razor [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+
+<MApp>
+    <MErrorHandler>
+        <MMain Class="fill-lighten-1">
+            <MRow Class="max-height ma-0">
+                <MCol Md=4 Sm=12 Class="neutral-lighten-5">
+                    <image class="ml-12 mt-12" style="height:40px;" src="https://cdn.masastack.com/stack/images/logo/MASAStack/logo-h-en.png?x-oss-process=image/resize,w_200"></image>
+                    <div class="d-flex pb-12" style="height: calc(100% - 88px);">
+                        <div style="width:450px;" class="ma-auto">
+                            <h3 class="neutral-lighten-1--text">让变化更简单</h3>
```
Comment (doddgu) [inline src/Web/Masa.Auth.Web.Sso/Shared/SideLayout.razor:12]: I18n?

File: src/Web/Masa.Auth.Web.Sso/wwwroot/i18n/zh-CN.json [hunk 7]
```diff
@@ -0,0 +1,33 @@
+{
```
Comment (doddgu) [inline src/Web/Masa.Auth.Web.Sso/wwwroot/i18n/zh-CN.json:1]: Pro?

**Review by doddgu — CHANGES_REQUESTED** [review]

File: src/Services/Masa.Auth.Service.Admin/Masa.Auth.Service.Admin.csproj [hunk 8]
```diff
@@ -12,9 +12,10 @@
 	</PropertyGroup>
 	
 	<ItemGroup>
+		<PackageReference Include="AspNetCore.HealthChecks.UI.Client" Version="6.0.4" />
```
[the remark this answers is not in the document]
Comment (MayueCif) [reply src/Services/Masa.Auth.Service.Admin/Masa.Auth.Service.Admin.csproj:15]: HealthCheck endpoint 。emo!  not currently

File: src/Web/Masa.Auth.Web.Sso/wwwroot/i18n/zh-CN.json (hunk 7 above)
Comment (MayueCif) [reply src/Web/Masa.Auth.Web.Sso/wwwroot/i18n/zh-CN.json:1]: i18n undo

File: src/Web/Masa.Auth.Web.Sso/Shared/SideLayout.razor (hunk 6 above)
Comment (MayueCif) [reply src/Web/Masa.Auth.Web.Sso/Shared/SideLayout.razor:12]: i18n undo

**Review by doddgu — APPROVED** [review]
LGTM

**Pull request merged**